core/trino-main/src/main/java/io/trino/connector/CoordinatorDynamicCatalogManager.java

```diff
@@ -125,11 +126,16 @@ public void loadInitialCatalogs()
                     executor,
                     catalogStore.getCatalogs().stream()
                             .map(catalog -> (Callable<?>) () -> {
-                                log.info("-- Loading catalog %s --", catalog.getCatalogHandle().getCatalogName());
-                                CatalogConnector newCatalog = catalogFactory.createCatalog(catalog);
-                                activeCatalogs.put(catalog.getCatalogHandle().getCatalogName(), newCatalog);
-                                allCatalogs.put(catalog.getCatalogHandle(), newCatalog);
-                                log.info("-- Added catalog %s using connector %s --", catalog.getCatalogHandle().getCatalogName(), catalog.getConnectorName());
+                                try {
+                                    CatalogConnector newCatalog = catalogFactory.createCatalog(catalog);
+                                    activeCatalogs.put(catalog.getCatalogHandle().getCatalogName(), newCatalog.getCatalog());
+                                    allCatalogs.put(catalog.getCatalogHandle(), newCatalog);
+                                    log.info("-- Added catalog %s using connector %s --", catalog.getCatalogHandle().getCatalogName(), catalog.getConnectorName());
+                                }
+                                catch (Exception e) {
```

Comment:
This can be `RuntimeException` since there are no checked exceptions. Or we should use `Throwable` if we want to catch everything.